pkg/logs/sender/batch_strategy.go

```diff
@@ -143,7 +143,7 @@ func (s *batchStrategy) flushBuffer(outputChan chan *message.Payload) {
 	if s.buffer.IsEmpty() {
 		if s.serverless {
 			// If there are no messages to send then mark the flush as done
-			s.flushDoneChan <- struct{}{}
+			go func() { s.flushDoneChan <- struct{}{} }()
```

Comment:
I assume this change is because we don't want to block when the channel is full, but it's not totally obvious why that would be the case. Might be worth including a comment with some explanation.